WcaOnRails/app/models/competition.rb

```diff
@@ -837,7 +839,14 @@ def has_fees?
   end
 
   def external_registration_page_required?
-    confirmed? && !use_wca_registration && created_at.present? && created_at > Date.new(2018, 12, 31)
+    confirmed? && !use_wca_registration && created_at.present? && is_modern_competition?
+  end
+
+  # This check will be used to avoid errors when updating old competitions
+  # Every time a new required field to the competition form is added, this date should be updated
+  # And this check should be added to function to see if it required
```

Comment:
Hmm, I don't think we can just increase this date over time though? My understanding is that: if field A is required at `2020-01-01` and we start requiring field B in `2022-01-01`, then we don't want to start validating field A from `2022-01-01` as it should be valid within these two years. Isn't that the case?

Reply:
As I see it, we only need to be validating future competitions. Only WCAT/Board/WST can edit the fields after announcement, if we are going to edit an old competition there will be a concrete reason. Many old competitions do not have certain fields filled out. 
Example, competition A is created in 2018, after some of the current validators are put in place. The competition was "hidden" (i.e. cancelled before that feature existed). This can longer be edited due to validator errors, though we would like to mark it correctly as cancelled (and it doesn't matter that it doesn't have all fields filled out). As this is an old competition, we should just forget about all the validators for it as there is a very specific reason it is being edited.
Competition B is announced in 2020, we later figure out that the state (i.e. in the city filed) of competition was listed incorrectly. We try to change it, but cannot as a new field is required but wasn't when the competition was announced (e.g. a deadline for changing registered events, as in Jacob's other PR). As this competition is now 'old' we should just ignore all validators and be allowed to change the city field and leave the new field blank.

To your concrete example, field A was validated when the competition was announced in e.g. 2021. Now when field B is required, all past competitions (so before 2022, if we were in 2022 now) is not being validated and it doesn't matter because it is an old competition, just that it is being validated for future competitions (i.e. the ones in 2022, if we were in 2022). When a field gets added in the form, we will be requiring them immediately, so there should not be any gap period for future competitions were validators are not used correctly. 

I hope my example was clear, otherwise follow up. The TLDR is that we only need to validators for WCAT to have clean (at the time of submission) submissions now and in the future, the past competitions should be possible to edit as needed without running into validator errors.